ui/platform_window/x11/x11_window_ozone.cc

```diff
@@ -93,10 +111,38 @@ uint32_t X11WindowOzone::DispatchEvent(const PlatformEvent& platform_event) {
   // (eg. double click) are broken.
   auto* ewpe = static_cast<EventWithPlatformEvent*>(platform_event);
   auto* event = static_cast<ui::Event*>(ewpe->event);
+
+  XID target = GetEventXWindow(ewpe);
+
+  // If the event came originally from another native window (was rerouted by
+  // PlatformEventSource), its location must be adapted using current window
+  // location.
+  if (target != xwindow())
+    ConvertEventLocationToCurrentWindowLocation(target, event);
+
   DispatchEventFromNativeUiEvent(
       event, base::Bind(&PlatformWindowDelegate::DispatchEvent,
                         base::Unretained(delegate())));
   return POST_DISPATCH_STOP_PROPAGATION;
 }
 
+void X11WindowOzone::OnLostCapture() {
+  delegate()->OnLostCapture();
+}
+
+void X11WindowOzone::ConvertEventLocationToCurrentWindowLocation(
```

Comment:
It would be nice to factor the core of this method out into a helper, so that it shares the logic with DesktopWindowTreeHostX11::ConvertEventToDifferentHost.

I believe we should add a TODO at least.